packages/deployer/internal/storage-ast-validator.js

```diff
@@ -89,12 +91,42 @@ class ModuleStorageASTValidator {
     return errors;
   }
 
-  findInvalidNamespaceMutations() {
-    logger.info(
-      'Unsafe storage namespace mutations are not yet validated in modules. Please only append to storage namespace structs.'
-    );
+  async findInvalidNamespaceMutations() {
+    const errors = [];
+
+    if (!this.previousAsts) {
+      return errors;
+    }
+
+    const previousStructsMap = await buildContractsStructMap(this.previousAsts);
+    const currentStructsMap = await buildContractsStructMap(this.asts);
+
+    const { modifications, removals } = compareStorageStructs({
```

Comment:
What about additions? These could be disruptive as well. E.g:
```
// :D
struct HappyStruct {
  address thing;
}
```
mutating to
```
// :(
struct HappyStruct {
  uint newSupahValue;
  address thing;
}
```

Reply:
The comparator is getting it right, but there's something failing when we get the current artifacts. This works the second time you run the deploy script.

I created a ticket for following up this issue.

Reply:
Additionally added a test to cover that situation in the comparator tests.